services/libssh/cve_2018_10933_server.py

```diff
@@ -0,0 +1,323 @@
+    xffffffff,
+    MSG_KEXINIT,
+    DEBUG,
+    ERROR,
+    WARNING,
+    MSG_IGNORE,
+    MSG_DISCONNECT,
+    MSG_DEBUG,
+    MSG_NAMES,
+    MSG_UNIMPLEMENTED,
+    MSG_USERAUTH_SUCCESS,
+    cMSG_UNIMPLEMENTED,
+)
+from paramiko.ssh_exception import SSHException
+from paramiko.packet import NeedRekeyException
+
+from consts import SERVER_SIG, EVENT_TYPE_FIELD_NAME, SSH_ALERT_TYPE, USERNAME_FIELD_NAME, PASSWORD_FIELD_NAME,\
+    ADDITIONAL_FIELDS_FIELD_NAME, KEY_FIELD_NAME, CVE_SSH_PORT
+
+# setup logging
+paramiko.util.log_to_file("demo_server.log")
+host_key = paramiko.RSAKey(filename=os.path.join(os.path.dirname(__file__), "test_rsa.key"))
+
+
+class CVETransport(paramiko.Transport):
+    alert = None
+    def __init__(self, *args, **kwargs):
+        super(CVETransport, self).__init__(*args, **kwargs)
+        self.local_version = SERVER_SIG
+
+    def run(self):
+        # (use the exposed "run" method, because if we specify a thread target
+        # of a private method, threading.Thread will keep a reference to it
+        # indefinitely, creating a GC cycle and not letting Transport ever be
+        # GC'd. it's a bug in Thread.)
+
+        # Hold reference to 'sys' so we can test sys.modules to detect
+        # interpreter shutdown.
+        self.sys = sys
+
+        # active=True occurs before the thread is launched, to avoid a race
+        _active_threads.append(self)
+        tid = hex(long(id(self)) & xffffffff)
+        if self.server_mode:
+            self._log(DEBUG, "starting thread (server mode): {}".format(tid))
+        else:
+            self._log(DEBUG, "starting thread (client mode): {}".format(tid))
+        try:
+            try:
+                self.packetizer.write_all(b(self.local_version + "\r\n"))
+                self._log(
+                    DEBUG,
+                    "Local version/idstring: {}".format(self.local_version),
+                )  # noqa
+                self._check_banner()
+                # The above is actually very much part of the handshake, but
+                # sometimes the banner can be read but the machine is not
+                # responding, for example when the remote ssh daemon is loaded
+                # in to memory but we can not read from the disk/spawn a new
+                # shell.
+                # Make sure we can specify a timeout for the initial handshake.
+                # Re-use the banner timeout for now.
+                self.packetizer.start_handshake(self.handshake_timeout)
+                self._send_kex_init()
+                self._expect_packet(MSG_KEXINIT)
+
+                while self.active:
+                    if self.packetizer.need_rekey() and not self.in_kex:
+                        self._send_kex_init()
+                    try:
+                        ptype, m = self.packetizer.read_message()
+                    except NeedRekeyException:
+                        continue
+                    if ptype == MSG_USERAUTH_SUCCESS:
```

Comment:
You should add a comment on the function header saying that that this code was copied from paramiko, and another comment for this line, saying that this if actually implements the detection of the exploit attempt of the vulnerability